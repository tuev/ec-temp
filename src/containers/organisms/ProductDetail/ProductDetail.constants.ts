export const CHANGE_COLOR = 'CHANGE_COLOR'
export const CHANGE_SIZE = 'CHANGE_SIZE'
export const CHANGE_QUANTITY = 'CHANGE_QUANTITY'
