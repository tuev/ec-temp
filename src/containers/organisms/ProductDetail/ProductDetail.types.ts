import {
  CHANGE_COLOR,
  CHANGE_SIZE,
  CHANGE_QUANTITY,
} from './ProductDetail.constants'
import { SIZE_PARAMS } from 'components/molecules/SizeSelect/SizeSelect.types'
import * as ProductItemInfoType from 'components/molecules/ProductItemInfo/ProductInfo.types'
import * as ProductItemBuyType from 'components/molecules/ProductItemBuy/ProductItemBuy.types'
import * as BreadcrumbType from 'components/atoms/Breadcrumbs/Breadcrumbs.types'
import { ColorType } from 'theme/colors'

export interface IProductDetailProps {
  breadcrumbLinks: BreadcrumbType.IBreadCrumbItem[]
  activeBreadcrumb: string
  images: string[]
  rating: number
  quantity: number
  colors: ColorType[]
  sizes: SIZE_PARAMS[]
  name: string
  price: string
  id: string
}

export interface IProductOptions {
  color: ColorType
  size: SIZE_PARAMS
  quantity: number
}

export interface IColorAction {
  type: typeof CHANGE_COLOR
  payload: ColorType
}

export interface ISizeAction {
  type: typeof CHANGE_SIZE
  payload: SIZE_PARAMS
}

export interface IQuantityAction {
  type: typeof CHANGE_QUANTITY
  payload: number
}

export type ProductSelectAction = IColorAction | ISizeAction | IQuantityAction

export type ProductSelectReducer = (
  optionValue: IProductOptions,
  action: ProductSelectAction
) => IProductOptions

export interface IHandleOptionsChange {
  color: (colorValue: ColorType) => void
  size: (sizeValue: SIZE_PARAMS) => void
  quantity: (quantityValue: number) => void
}

export type ProductSelectHook = (
  initValue: IProductOptions
) => [IProductOptions, IHandleOptionsChange]

export type HandleBuyProduct = ({
  id,
  color,
  size,
  quantity,
}: { id: string } & IProductOptions) => unknown

export type ProductDetailProps = IProductDetailProps & {
  handleBuy?: HandleBuyProduct
}
