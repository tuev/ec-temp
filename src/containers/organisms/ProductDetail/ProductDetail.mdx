# Stories mdx for ProductDetail

import { Meta, Story } from '@storybook/addon-docs/blocks'
import * as stories from './ProductDetail.stories.tsx'
import ProductDetail from './ProductDetail.tsx'

<Story name="basic" />