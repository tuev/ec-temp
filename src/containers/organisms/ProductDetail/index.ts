/**
 *
 * Index for ProductDetail
 *
 */

import ProductDetail from './ProductDetail'

export default ProductDetail
