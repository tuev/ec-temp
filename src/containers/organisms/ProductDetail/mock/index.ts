import { ProductDetailProps } from './../ProductDetail.types'
import img1 from './img2.jpg'
import img2 from './img3.jpg'
import img3 from './img4.jpg'

export const itemInfo: ProductDetailProps = {
  id: '123468761283',
  images: [img1, img2, img3],
  breadcrumbLinks: [{ href: '/', title: 'Home' }],
  activeBreadcrumb: 'product',
  rating: 1,
  name: 'Product name',
  price: '2$',
  colors: ['black', 'blue', 'red', 'yellow', 'green'],
  sizes: ['L', 'M', 'S', 'XL'],
  quantity: 1,
}
