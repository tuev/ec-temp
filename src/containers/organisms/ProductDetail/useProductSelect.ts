import { useCallback, useReducer, useMemo } from 'react'

import * as ACTION_TYPE from './ProductDetail.constants'
import { SIZE_PARAMS } from 'components/molecules/SizeSelect/SizeSelect.types'

import { ColorType } from 'theme/colors'
import { ProductSelectHook, ProductSelectReducer } from './ProductDetail.types'

export const reducer: ProductSelectReducer = (initState, action) => {
  switch (action.type) {
    case ACTION_TYPE.CHANGE_COLOR:
      return {
        ...initState,
        color: action.payload,
      }
    case ACTION_TYPE.CHANGE_SIZE:
      return {
        ...initState,
        size: action.payload,
      }
    case ACTION_TYPE.CHANGE_QUANTITY:
      return {
        ...initState,
        quantity: action.payload,
      }
    default:
      return initState
  }
}

const useProductSelect: ProductSelectHook = initState => {
  const [options, dispatch] = useReducer(reducer, initState)
  const onColorChange = useCallback(
    (color: ColorType) => dispatch({ type: 'CHANGE_COLOR', payload: color }),
    []
  )
  const onSizeChange = useCallback(
    (size: SIZE_PARAMS) =>
      dispatch({ type: ACTION_TYPE.CHANGE_SIZE, payload: size }),
    []
  )
  const onQuantityChange = useCallback(
    (quantity: number) =>
      dispatch({ type: ACTION_TYPE.CHANGE_QUANTITY, payload: quantity }),
    []
  )

  const onOptionsChange = useMemo(
    () => ({
      color: onColorChange,
      quantity: onQuantityChange,
      size: onSizeChange,
    }),
    [onColorChange, onQuantityChange, onSizeChange]
  )

  return [options, onOptionsChange]
}

export default useProductSelect
