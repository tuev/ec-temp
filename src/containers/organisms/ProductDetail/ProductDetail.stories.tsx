/**
 *
 *  Stories for ProductDetail
 *
 */

import React from 'react'
import { action } from '@storybook/addon-actions'
import ProductDetail from './ProductDetail'
import { itemInfo } from './mock'

export default {
  component: ProductDetail,
  includeStories: ['basic'],
  title: 'ProductDetail',
}

export const basic = () => <ProductDetail {...itemInfo} />
