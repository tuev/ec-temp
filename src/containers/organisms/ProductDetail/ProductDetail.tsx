/**
 *
 * ProductDetail
 *
 */

import React, { FC, useCallback } from 'react'
import { Grid } from '@material-ui/core'
import ProductImageSlider from 'components/molecules/ProductImageSlider'
import Wrapper from 'components/atoms/Wrapper'
import Breadcrumbs from 'components/atoms/Breadcrumbs'
import Typography from 'components/atoms/Typography'

import Rating from 'components/atoms/Rating'
import { ProductDetailProps } from './ProductDetail.types'
import ColorSelect from 'components/molecules/ColorSelect'
import SizeSelect from 'components/molecules/SizeSelect'
import QuantityInput from 'components/molecules/QuantityInput'
import Button from 'components/atoms/Button'

import useProductSelect from './useProductSelect'

const ProductDetail: FC<ProductDetailProps> = (props: ProductDetailProps) => {
  const {
    id,
    images = [],
    breadcrumbLinks = [],
    activeBreadcrumb = 'product',
    rating = 1,
    name = '',
    price = '',
    colors = [],
    sizes = [],
    quantity = 1,
    handleBuy,
  }: ProductDetailProps = props

  const [options, onOptionsChange] = useProductSelect({
    color: colors[0],
    size: sizes[0],
    quantity,
  })

  const handleBuyClick = useCallback(() => {
    if (handleBuy) {
      handleBuy({
        id,
        ...options,
      })
    }
  }, [handleBuy, id, options])

  return (
    <Grid container={true}>
      <Grid item={true} xs={12} md={4}>
        <Wrapper p="1rem">
          <ProductImageSlider images={images} />
        </Wrapper>
      </Grid>
      <Grid item={true} xs={12} md={8}>
        <Wrapper p="1rem">
          <Wrapper mb="1rem">
            <Breadcrumbs links={breadcrumbLinks} active={activeBreadcrumb} />
          </Wrapper>
          <Rating value={rating} />

          <Wrapper>
            <Wrapper>
              <Typography customvariant="header2">{name}</Typography>
              <Typography customvariant="header2" customcolor="red">
                {price}
              </Typography>
            </Wrapper>

            <Wrapper py="0.5rem" mt="1rem">
              <Typography customvariant="header3" customcolor="black1">
                Color
              </Typography>
              <ColorSelect
                colors={colors}
                value={options.color}
                onChange={onOptionsChange.color}
              />
            </Wrapper>

            <Wrapper py="0.5rem">
              <Typography customvariant="header3" customcolor="black1">
                Size
              </Typography>
              <SizeSelect
                sizes={sizes}
                value={options.size}
                onChange={onOptionsChange.size}
              />
            </Wrapper>

            <Wrapper py="0.5rem">
              <Typography customvariant="header3" customcolor="black1">
                Quantity
              </Typography>
              <QuantityInput
                value={options.quantity}
                onChange={onOptionsChange.quantity}
              />
            </Wrapper>

            <Wrapper py="1rem">
              <Button
                onClick={handleBuyClick}
                data-testid="product-detail-buy-btn"
              >
                Add to cart
              </Button>
            </Wrapper>
          </Wrapper>
        </Wrapper>
      </Grid>
    </Grid>
  )
}

export default ProductDetail
