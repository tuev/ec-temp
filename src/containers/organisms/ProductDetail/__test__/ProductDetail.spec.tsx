/**
 *
 * Tests for ProductDetail
 *
 */

import React from 'react'

import { render, fireEvent } from '@testing-library/react'
import ProductDetail from '..'

describe('App ProductDetail', () => {
  it('Test get default ProductDetail no params ', () => {
    // test snapshot
    const wrapper = render(<ProductDetail />)
    expect(wrapper).toMatchSnapshot()
  })
})
