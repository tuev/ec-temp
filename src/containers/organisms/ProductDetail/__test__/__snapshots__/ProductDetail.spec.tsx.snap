// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App ProductDetail Test get default ProductDetail no params  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c1 {
  width: 320px;
  height: 320px;
  position: relative;
  overflow: hidden;
}

.c0 {
  padding: 1rem;
}

.c2 {
  margin-top: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  margin-bottom: 1rem;
}

.c7 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  margin-top: 1rem;
}

.c9 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c11 {
  padding-top: 1rem;
  padding-bottom: 1rem;
}

.c4 {
  color: #2E2E2E;
}

.c5 {
  font-size: 24px !important;
  line-height: 1.5 !important;
  font-family: 'Montserrat Alternates' !important;
  font-weight: bold !important;
  color: #2E2E2E;
}

.c6 {
  font-size: 24px !important;
  line-height: 1.5 !important;
  font-family: 'Montserrat Alternates' !important;
  font-weight: bold !important;
  color: #F62F5E;
}

.c8 {
  font-size: 16px !important;
  line-height: 1.5 !important;
  font-family: 'Montserrat Alternates' !important;
  font-weight: bold !important;
  color: #6C6C6C;
}

.c12 {
  min-width: 163px !important;
  min-height: 48px !important;
  font-size: 14px;
  border-radius: 32px;
}

<body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
        >
          <div
            class="c0"
          >
            <div
              style="width: 320px;"
            >
              <div
                class="c1"
              />
              <div
                class="c2"
                display="flex"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
        >
          <div
            class="c0"
          >
            <div
              class="c3"
            >
              <nav
                aria-label="breadcrumb"
                class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
              >
                <ol
                  class="MuiBreadcrumbs-ol"
                >
                  <li
                    class="MuiBreadcrumbs-li"
                  >
                    <p
                      class="MuiTypography-root c4 MuiTypography-body1"
                    >
                      product
                       
                    </p>
                  </li>
                </ol>
              </nav>
            </div>
            <span
              class="MuiRating-root "
            >
              <label
                class="MuiRating-label"
                for="app-rating-1"
              >
                <span
                  class="MuiRating-icon MuiRating-iconFilled"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiRating-visuallyhidden"
                >
                  1 Star
                </span>
              </label>
              <input
                checked=""
                class="MuiRating-visuallyhidden"
                id="app-rating-1"
                name="app-rating"
                type="radio"
                value="1"
              />
              <label
                class="MuiRating-label"
                for="app-rating-2"
              >
                <span
                  class="MuiRating-icon MuiRating-iconEmpty"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiRating-visuallyhidden"
                >
                  2 Stars
                </span>
              </label>
              <input
                class="MuiRating-visuallyhidden"
                id="app-rating-2"
                name="app-rating"
                type="radio"
                value="2"
              />
              <label
                class="MuiRating-label"
                for="app-rating-3"
              >
                <span
                  class="MuiRating-icon MuiRating-iconEmpty"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiRating-visuallyhidden"
                >
                  3 Stars
                </span>
              </label>
              <input
                class="MuiRating-visuallyhidden"
                id="app-rating-3"
                name="app-rating"
                type="radio"
                value="3"
              />
              <label
                class="MuiRating-label"
                for="app-rating-4"
              >
                <span
                  class="MuiRating-icon MuiRating-iconEmpty"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiRating-visuallyhidden"
                >
                  4 Stars
                </span>
              </label>
              <input
                class="MuiRating-visuallyhidden"
                id="app-rating-4"
                name="app-rating"
                type="radio"
                value="4"
              />
              <label
                class="MuiRating-label"
                for="app-rating-5"
              >
                <span
                  class="MuiRating-icon MuiRating-iconEmpty"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiRating-visuallyhidden"
                >
                  5 Stars
                </span>
              </label>
              <input
                class="MuiRating-visuallyhidden"
                id="app-rating-5"
                name="app-rating"
                type="radio"
                value="5"
              />
            </span>
            <div
              class=""
            >
              <div
                class=""
              >
                <p
                  class="MuiTypography-root c5 MuiTypography-body1"
                  customvariant="header2"
                />
                <p
                  class="MuiTypography-root c6 MuiTypography-body1"
                  customcolor="red"
                  customvariant="header2"
                />
              </div>
              <div
                class="c7"
              >
                <p
                  class="MuiTypography-root c8 MuiTypography-body1"
                  customcolor="black1"
                  customvariant="header3"
                >
                  Color
                </p>
                <div />
              </div>
              <div
                class="c9"
              >
                <p
                  class="MuiTypography-root c8 MuiTypography-body1"
                  customcolor="black1"
                  customvariant="header3"
                >
                  Size
                </p>
                <div
                  class=""
                />
              </div>
              <div
                class="c9"
              >
                <p
                  class="MuiTypography-root c8 MuiTypography-body1"
                  customcolor="black1"
                  customvariant="header3"
                >
                  Quantity
                </p>
                <div
                  class="c10"
                  display="flex"
                >
                  <button
                    class="MuiButtonBase-root MuiFab-root MuiFab-sizeSmall MuiFab-secondary"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiFab-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 13H5v-2h14v2z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                  <div
                    class="MuiFormControl-root MuiTextField-root"
                  >
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input"
                        min="0"
                        type="number"
                        value="1"
                      />
                      <fieldset
                        aria-hidden="true"
                        class="PrivateNotchedOutline-root-255 MuiOutlinedInput-notchedOutline"
                        style="padding-left: 8px;"
                      >
                        <legend
                          class="PrivateNotchedOutline-legend-256"
                          style="width: 0px;"
                        >
                          <span>
                            ​
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                  <button
                    class="MuiButtonBase-root MuiFab-root MuiFab-sizeSmall MuiFab-secondary"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiFab-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
              <div
                class="c11"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained c12 MuiButton-containedPrimary"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Add to cart
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-container"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
      >
        <div
          class="sc-AykKG BHWeQ"
        >
          <div
            style="width: 320px;"
          >
            <div
              class="sc-AykKE cxWcju"
            />
            <div
              class="sc-AykKG kXyjoA"
              display="flex"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
      >
        <div
          class="sc-AykKG BHWeQ"
        >
          <div
            class="sc-AykKG kryvSJ"
          >
            <nav
              aria-label="breadcrumb"
              class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
            >
              <ol
                class="MuiBreadcrumbs-ol"
              >
                <li
                  class="MuiBreadcrumbs-li"
                >
                  <p
                    class="MuiTypography-root sc-AykKH jvoJCj MuiTypography-body1"
                  >
                    product
                     
                  </p>
                </li>
              </ol>
            </nav>
          </div>
          <span
            class="MuiRating-root sc-AykKI ctvLEX"
          >
            <label
              class="MuiRating-label"
              for="app-rating-1"
            >
              <span
                class="MuiRating-icon MuiRating-iconFilled"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyhidden"
              >
                1 Star
              </span>
            </label>
            <input
              checked=""
              class="MuiRating-visuallyhidden"
              id="app-rating-1"
              name="app-rating"
              type="radio"
              value="1"
            />
            <label
              class="MuiRating-label"
              for="app-rating-2"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyhidden"
              >
                2 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyhidden"
              id="app-rating-2"
              name="app-rating"
              type="radio"
              value="2"
            />
            <label
              class="MuiRating-label"
              for="app-rating-3"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyhidden"
              >
                3 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyhidden"
              id="app-rating-3"
              name="app-rating"
              type="radio"
              value="3"
            />
            <label
              class="MuiRating-label"
              for="app-rating-4"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyhidden"
              >
                4 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyhidden"
              id="app-rating-4"
              name="app-rating"
              type="radio"
              value="4"
            />
            <label
              class="MuiRating-label"
              for="app-rating-5"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
              <span
                class="MuiRating-visuallyhidden"
              >
                5 Stars
              </span>
            </label>
            <input
              class="MuiRating-visuallyhidden"
              id="app-rating-5"
              name="app-rating"
              type="radio"
              value="5"
            />
          </span>
          <div
            class="sc-AykKG kLLdqP"
          >
            <div
              class="sc-AykKG kLLdqP"
            >
              <p
                class="MuiTypography-root sc-AykKH dZvxdr MuiTypography-body1"
                customvariant="header2"
              />
              <p
                class="MuiTypography-root sc-AykKH bWMmxA MuiTypography-body1"
                customcolor="red"
                customvariant="header2"
              />
            </div>
            <div
              class="sc-AykKG bwAPKj"
            >
              <p
                class="MuiTypography-root sc-AykKH eHeTJy MuiTypography-body1"
                customcolor="black1"
                customvariant="header3"
              >
                Color
              </p>
              <div />
            </div>
            <div
              class="sc-AykKG bYCFOX"
            >
              <p
                class="MuiTypography-root sc-AykKH eHeTJy MuiTypography-body1"
                customcolor="black1"
                customvariant="header3"
              >
                Size
              </p>
              <div
                class="sc-fzXfLO iVXeHU"
              />
            </div>
            <div
              class="sc-AykKG bYCFOX"
            >
              <p
                class="MuiTypography-root sc-AykKH eHeTJy MuiTypography-body1"
                customcolor="black1"
                customvariant="header3"
              >
                Quantity
              </p>
              <div
                class="sc-AykKG gfmPnj"
                display="flex"
              >
                <button
                  class="MuiButtonBase-root MuiFab-root MuiFab-sizeSmall MuiFab-secondary"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiFab-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 13H5v-2h14v2z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <div
                  class="MuiFormControl-root MuiTextField-root"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      min="0"
                      type="number"
                      value="1"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-255 MuiOutlinedInput-notchedOutline"
                      style="padding-left: 8px;"
                    >
                      <legend
                        class="PrivateNotchedOutline-legend-256"
                        style="width: 0px;"
                      >
                        <span>
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
                <button
                  class="MuiButtonBase-root MuiFab-root MuiFab-sizeSmall MuiFab-secondary"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiFab-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            <div
              class="sc-AykKG ibYTzz"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-AykKK jPLyQf MuiButton-containedPrimary"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Add to cart
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
